### ISSUE TITLE
inbox style should return array of notifications object not last notification

### ISSUE THREAD

**Issue by abdoljabbar** [phonegap/phonegap-plugin-push#314]
hi,
First thank u for this awesome Plugin.
when i send notifications with inbox style and tap on them,   push.on('notification', callBack(data)) return just last notification object.
I think it should be array of notifications object not just last.
Tnx in advance.

**Comment by nicoabie** [comment]
+1
There are two approaches, to return always an array of notifications on the onNotification callback or call it multiple times for each item in the cached notification list.
Regarding that list that following the current approach would be stored in memory, I thought of storing it in sqlite, is it there any other place to potentially store those cached notifications?

**Comment by fredgalvao** [comment]
If you're talking about web code (not native), you can also use localstorage as it's more widely supported than sqlite. Now, if you meant native code, you can use whatever storage android/ios has available, although third party dependencies are avoided.

**Comment by nicoabie** [comment]
Thanks @fredgalvao, will try to submit a pull request for the in memory array caching first and then investigate the native storage options without using any third party dependencies.

**Comment by fredgalvao** [comment]
The cache of these notifications is already done in memory, check this [HashMap](https://github.com/phonegap/phonegap-plugin-push/blob/master/src/android/com/adobe/phonegap/push/GCMIntentService.java#L41).

**Comment by nicoabie** [comment]
Yes but I think the same should be done for the Bundle of extras where it is possible now to loose data (last cached bundle deletes the previous one) [shoudn't this be an array too](https://github.com/phonegap/phonegap-plugin-push/blob/master/src/android/com/adobe/phonegap/push/PushPlugin.java#L191)?

**Comment by fredgalvao** [comment]
That's later on the pipeline. If you change what a "bundle extra" means at the `GCMIntentService` so that it possibly includes multiple notification objects for a single `notId`, that single `extra` Bundle object will already contain everything you need, and will be converted to json properly.

**Comment by nicoabie** [comment]
I see what you mean but I thought something like this in the meantime https://github.com/phonegap/phonegap-plugin-push/pull/400

**Comment by fredgalvao** [comment]
Although that seems like it would "work", taking into consideration @macdonst 's observations on the code, I still think making the `Bundle extras` object hold all the notifications by itself is a better option. Saves a lot of code at the PushPlugin level, and avoids pretty much every concern stated by @macdonst.

**Comment by macdonst** [comment]
Upon further review of this one I'm not going to make any changes. Having multiple notification events fire when you click on one notification is not idea, neither is adding all the data into one giant notifications especially since the older data may no longer need to be processed.

Instead, I'm recommending that you use the "content available" flag when 1.5.x is out.

**Comment by pallaviMN** [comment]
Hi Team,

Can you please tell the fix for it. Which files need to be changed & what? GCMintentService or  PushPlugin.java?

**Comment by purinagripal** [comment]
Hi,

I'm trying to store each notification in localStorage using content-available = 1 but is not working. The notification event handler is fire when you click on a notification and the app is launch, but not while the app is on background, so I get only the last notification in localStorage.

If the app is foreground I get all the notification in localStorage, but I need the same behavior with app on background too.

I'm trying to store each notification on background because I need the list of new events when I open the app. How can I get that?

My notification event handler:
`push.on('notification', function(data) {
            
            console.log('notificacion: '+data.message);
            console.log('id_evento: '+data.additionalData.id_evento);
            console.log(data.additionalData);
            
            var notif_anteriores;
            
            if( window.localStorage.getItem('notificaciones') ) {
                console.log('existe localstorage');
                // tenemos notificaciones guardadas
                notif_anteriores = window.localStorage.getItem('notificaciones');
                notif_anteriores = JSON.parse(notif_anteriores);
                // añadimos la nueva
                notif_anteriores.push(data.additionalData.id_evento);
                
                window.localStorage.setItem('notificaciones', JSON.stringify(notif_anteriores));
                
            } else {
                console.log('crea localstorage con '+data.message);
                // aun no hay ninguna notificacion guardada
                notif_anteriores = [data.additionalData.id_evento];
                // guardamos la primera
                window.localStorage.setItem( 'notificaciones', JSON.stringify(notif_anteriores) );
            }
            
        });`

This is the notification payload:
`$payload = array(
            'title'      => 'Titulo', // Titulo de la notificación 
            'message'     => $mensaje, //mensaje a enviar 
            'id_evento' => $evento['id_evento'],
            
            'style' => "inbox",
            'summaryText' => "%n% nuevos eventos",
            
            'soundname' => '',
            'vibrationPattern' => [0],
            
            'content-available' => '1'
        );
                
        $notificacion = array(
            'registration_ids' => $registrationIds,
            'data' => $payload
        );`

thank you
